src/App.tsx

```diff
@@ -1,8 +1,98 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { getTodos, getUsers } from './helper/api';
+import { SortPanel } from './components/SortPanel';
+import { TodoList } from './components/TodoList';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+const normalizeTodos = async () => {
+  const todos = await getTodos();
+  const users = await getUsers();
+
+  return todos.map((todo: Todo) => {
+    return {
+      ...todo,
+      user: users.find((user: User) => user.id === todo.userId),
+    };
+  });
+};
+
+const App = () => {
+  const [loading, setLoading] = useState(false);
+  const [isContentVisible, setContentVisible] = useState(false);
+  const [todos, setTodos] = useState<TodoNormalized[]>([]);
+
+  const onLoadClick = () => {
+    setLoading(true);
+    normalizeTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+        setLoading(false);
+        setContentVisible(true);
+      });
+  };
+
+  const sortTodos = (sortName: string|boolean) => {
+    switch (sortName) {
+      case 'title':
+        setTodos([...todos].sort((a, b) => {
+          return a.title.localeCompare(b.title);
+        }));
+
+        return;
+      case 'completed':
+        setTodos([...todos].sort((a, b) => {
+          return +a.completed - +b.completed;
+        }));
+
+        return;
+      case 'user':
+        setTodos([...todos].sort((a, b) => {
+          return a.user.name.localeCompare(b.user.name);
+        }));
+
+        return;
+      default:
+        setTodos([...todos]);
+    }
+  };
+
+
+  const changeTodoStatus = (id: number) => {
+    const actualTodos = todos.map(todo => {
+      if (todo.id === id) {
+        return { ...todo,
+          completed: !todo.completed };
```

Comment:
```suggestion
        return { 
          ...todo,
          completed: !todo.completed 
        };
```
this is how I showed you before